lib/Core/SpecialFunctionHandler.cpp

```diff
@@ -1180,26 +1185,22 @@ void SpecialFunctionHandler::handleIsPersisted(ExecutionState &state,
     bool success;
     assert(state.addressSpace.resolveOne(state, executor.solver, ptrExpr, res, success));
     assert(success && "could not resolve one! (isPersisted)");
-    pmemObjs.push_back(res);
-  }
-  
-  std::unordered_set<std::string> errors;
-  std::unordered_set<const MemoryObject *> mos;
-  bool isPersisted = true;
-  for (ObjectPair &op : pmemObjs) {
-    const MemoryObject *mo = op.first; 
-    mos.insert(mo);
-    const ObjectState *os = op.second;
-    assert(isa<PersistentState>(os) && "trying to check if non-pmem is persisted!");
 
-    bool objPersisted = executor.isObjectPersisted(state, mo, errors);
-    isPersisted = isPersisted && objPersisted;
+    const MemoryObject *mo = res.first;
+    PersistentState *ps = dyn_cast<PersistentState>(state.addressSpace.getWriteable(mo, res.second));
+    assert(ps && "trying to check if non-pmem is persisted!");
+
+    bool hasErrs = executor.getPersistenceErrors(state, mo, errors);
+    emitErrs = emitErrs || hasErrs;
+    ps->flushAll();
```

Comment:
Having this here instead of in some shared Executor function is the same thing that bit us in the butt earlier today, with SpecialFunctionHandler checking persistence differently than Executor.

Reply:
Never mind, I see we haven't added it to Executor yet.

On another note, I actually still don't remain 100% convinced that we wouldn't mask anything by flushing all writes. Definitely okay with this for now though.